app/views/home/index.html.erb

```diff
@@ -1,21 +1,80 @@
-<header>
-  <title>HackerNews</title>
-</header>
-
-<body>
-  <div class="page-header">
-    <%= link_to 'Hacker News', root_path, class: 'page-title' %> |
-    <%= link_to 'new', root_path, class: 'link' %> |
-    <%= link_to 'comments', root_path, class: 'link' %>
-
-    <% if current_user.present? %>
-      | <%= link_to 'submit', new_link_path, class: 'link' %>
-      <%= link_to 'edit account', edit_user_path(current_user), class: 'link right-header' %>
-      <span class="right-header"> | </span>
-      <%= link_to 'logout', logout_path, method: :delete, class: 'link right-header' %>
-    <% else %>
-      <%= link_to 'login', login_path, class: 'link right-header' %>
+<div class="page-header">
+  <%= link_to 'Hacker News', root_path, class: 'page-title' %> |
+  <%= link_to 'new', root_path, class: 'link' %> |
+  <%= link_to 'comments', root_path, class: 'link' %>
+
+  <% if current_user.present? %>
+    | <%= link_to 'submit', new_link_path, class: 'link' %>
+    <%= link_to 'edit account', edit_user_path(current_user), class: 'link right-header' %>
+    <span class="right-header"> | </span>
+    <%= link_to 'logout', logout_path, method: :delete, class: 'link right-header' %>
+  <% else %>
+    <%= link_to 'login', login_path, class: 'link right-header' %>
+  <% end %>
+</div>
+
+<ul class="list-group">
+
+  <% if @links.present? %>
+  <% @display_counter = 0 %>
+    <% @links.each do |link| %>
+      <% @display_counter += 1 %>
+
+      <li class="list-group-item" id="link-<%= "#{link.id}" %>">
+        <table>
+          <tr>
+            <td class="rank"><%= "#{@display_counter}" %></td>
+
+            <td>
+            <!-- If user is not logged in, cannot vote -->
+            <% if current_user.present? %>
+              <!-- Check if user has already voted -->
+              <% voteLookup = HackerNews::VoteLookupService.new(current_user, link) %>
```

Comment:
I'm thinking instead of assigning a variable here, we could somehow accomplish a similar thing in the controller with (and I'm just pseudo-coding here to demonstrate):

```
@user_voted_links = if @current_user
  @current_user.votes.map(&:links).flatten
else
  []
end
```

then here, we could do something like:

```
if link.in?(@user_voted_links)
  ... do a thing
```

does that make sense?